src/Symfony/Component/Console/Attribute/AsLockedCommand.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Attribute;
+
+#[\Attribute(\Attribute::TARGET_CLASS)]
+class AsLockedCommand
+{
+    /**
+     * @param string            $name        The name of the command, used when calling it (i.e. "cache:clear")
+     * @param string|null       $description The description of the command, displayed with the help page
+     * @param string|bool|array $lock        The string used as the key to create lock, or a boolean to use the command's name, or callable which returns a string
+     * @param bool              $blocking    Is this lock should be blocking ? true to wait until the lock is available, false otherwise
+     * @param string[]          $aliases     The list of aliases of the command. The command will be executed when using one of them (i.e. "cache:clean")
+     * @param bool              $hidden      If true, the command won't be shown when listing all the available commands, but it can still be run as any other command
+     */
+    public function __construct(
+        public string $name,
+        public ?string $description = null,
```

Comment:
We have those fields on the `AsCommand` attribute already. why do we duplicate them?

Reply:
The idea was to not have to use both attributes, but now I realize it was probably not appropriate. I've removed the duplicate properties in the `AsLockedCommand`, so each attribute add his own behavior.